### DISCUSSION TITLE
Can Mall be integrated into existing an October CMS website with an existing products database table?

### DISCUSSION THREAD

**Discussion by resdd** [OFFLINE-GmbH/oc-mall-plugin#1052] [Q&A]
I built a website for one of my clients several years ago using October CMS and a custom-designed products database. It's essentially a product website without any e-commerce capability (at the time I build it, my client didn't want to do online sales). In addition to the existing database tables for products and categories, I also have existing page templates for displaying products by category and product details. Now my client needs to add e-commerce capabilities to the website.

Is it possible to install the Mall plugin and use my existing product data table, adding additional fields required by Mall, and then integrate the necessary form elements into my existing product page templates? I'm hoping that I can find an e-commerce solution that is customizable enough to be easily integrated into the existing website without needing to overhaul the product data structure and customized navigation and search capabilities that currently exist.

Thanks to any help. I realize more info may be needed, so please ask whatever questions are necessary.

**Comment by tobias-kuendig** [comment]
> Is it possible to install the Mall plugin and use my existing product data table, adding additional fields required by Mall, and then integrate the necessary form elements into my existing product page templates? 

That would certainly be possible but might not be the best solution as you would end up with a Frankenstein setup. If you really want to adopt another e-commerce solution, I suggest you migrate your data into the proper structure that the plugin requires and then use the plugin as intended. 

Maybe our https://github.com/OFFLINE-GmbH/oc-microcart-plugin plugin is of interest to you. It only provides a Cart and a Checkout component but you have to bring your own products. It is by no means as fleshed out as Mall but since you already have a shop structure in place it might be a good fit.

**Reply by resdd** [reply]
Thank you for your reply, Tobias. I did look at the MicroCart plugin as well and I'm working through the process of whether Mall or MicroCart will be the best solution for my situation.

I need to integrate e-commerce into the website as quickly as possible, so I'm looking for a solution that will result in the least amount of work. It seems that if I were to use MicroCart, I'd need to write some code to handle inventory tracking and shipping calculations. While I am capable of doing that, it seems like it would extend the time-frame of the project longer than I'd like to.

If I go with Mall, it seems that the work will primarily be data migration, configuration of payment gateway and shipping rates, and possibly custom styling of the product pages. This seems to me like the "lighter" amount of work to do and something more reasonable for an as-soon-as-possible turnaround.

Both plugins look like great options, thank you for your excellent work!. I'm setting up a staging copy of the website to test them and see evaluate the work needed. If I have more questions along the way, would you mind answering them here?

**Reply by tobias-kuendig** [reply]
> would you mind answering them here?

I can't make any promises as this is highly dependent on my workload ;) But feel free to post them, if I've got the time I will try to answer. Or maybe someone else from the community can chime in.
